src/CoreBundle/Controller/PermissionController.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+declare(strict_types=1);
+
+/* For licensing terms, see /license.txt */
+
+namespace Chamilo\CoreBundle\Controller;
+
+use Chamilo\CoreBundle\Entity\PermissionRelRole;
+use Chamilo\CoreBundle\Form\PermissionType;
+use Chamilo\CoreBundle\Repository\PermissionRelRoleRepository;
+use Chamilo\CoreBundle\Repository\PermissionRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Security\Http\Attribute\IsGranted;
+
+class PermissionController extends AbstractController
+{
+
+    #[IsGranted('ROLE_ADMIN')]
+    #[Route('/permissions/test', name: 'permissions_test')]
+    public function testPermissions(): Response
+    {
+        // Test roles and permission slug
+        $roles = ['ROLE_STUDENT', 'ROLE_TEACHER'];
+        $permissionSlug = 'analytics:view';
+
+        // Call the api_get_permission function and log the result
+        $hasPermission = api_get_permission($permissionSlug, $roles);
+        error_log('Permission check for ' . $permissionSlug . ' with roles ' . implode(', ', $roles) . ': ' . ($hasPermission ? 'true' : 'false'));
+
+        // Return a simple response for testing purposes
+        return new Response('<html><body>Permission check result: ' . ($hasPermission ? 'true' : 'false') . '</body></html>');
+    }
+
+    #[IsGranted('ROLE_ADMIN')]
+    #[Route('/permissions', name: 'permissions')]
+    public function index(
+        PermissionRepository $permissionRepo,
+        PermissionRelRoleRepository $permissionRelRoleRepo,
+        Request $request,
+        EntityManagerInterface $em
+    ): Response {
+        $permissions = $permissionRepo->findAll();
+        $roles = ['ROLE_INVITEE', 'ROLE_STUDENT', 'ROLE_TEACHER', 'ROLE_ADMIN', 'ROLE_SUPER_ADMIN', 'ROLE_GLOBAL_ADMIN', 'ROLE_RRHH', 'ROLE_QUESTION_MANAGER', 'ROLE_SESSION_MANAGER', 'ROLE_STUDENT_BOSS'];
```

Comment:
We should have a method somewhere that returns all supported roles. Having this list hardcoded here is probably not a good idea.